cmd/cofidectl/cmd/trustzone/trustzone.go

```diff
@@ -215,6 +216,58 @@ func (c *TrustZoneCommand) GetAddCommand() *cobra.Command {
 	return cmd
 }
 
+var trustZoneDelCmdDesc = `
+This command will delete a trust zone from the Cofide configuration state.
+`
+
+func (c *TrustZoneCommand) GetDelCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "del [NAME]",
+		Short: "Delete a trust zone",
+		Long:  trustZoneDelCmdDesc,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.deleteTrustZone(cmd.Context(), args[0])
+		},
+	}
+	return cmd
+}
+
+func (c *TrustZoneCommand) deleteTrustZone(ctx context.Context, name string) error {
+	ds, err := c.cmdCtx.PluginManager.GetDataSource(ctx)
+	if err != nil {
+		return err
+	}
+
+	clusters, err := ds.ListClusters(name)
+	if err != nil {
+		return err
+	}
+
+	// Fail if any clusters in the trust zone are up.
+	for _, cluster := range clusters {
+		if deployed, err := helmprovider.IsClusterDeployed(ctx, cluster); err != nil {
+			return err
+		} else if deployed {
+			return fmt.Errorf("cluster %s in trust zone %s cannot be deleted while it is up", cluster.GetName(), name)
+		}
+	}
+
+	for _, cluster := range clusters {
+		err = ds.DestroyCluster(cluster.GetName(), name)
+		if err != nil {
+			return fmt.Errorf("failed to destroy cluster %s: %w", cluster.GetName(), err)
```

Comment:
In the event this errors partway through a delete, I think we'll have a partial delete case. Is there anything we can do to make this more transactional, or do we think that's premature for the moment?

Reply:
Good thinking, I'll come up with something.

Reply:
Rollback behaviour looks good, do we want to add some light test coverage to verify it works as expected in the event of a mid-destroy error? We can roll this into another branch if were wanting to get these merged to unblock

Reply:
Added some tests.